### PR TITLE
Disabled CollisionShape2D is not processed now

### LINKED ISSUE #24183: CollisionShape2D Disabled Check Problem
**Godot version:**

Godot master build

**OS/device including version:**

Ubuntu 18.04 / Radeon R7 250X

**Issue description:**

When I add CollisionShape2D(checked disabled feature) to RigidBody2D as a child, the RigidBody2D does not appear on play screen. If I uncheck CollisionShape2D disabled feature, it works correctly.

**Steps to reproduce:**
1. Create a new scene
2. Add RigidBody2D as a root
3. Add CollisionShape2D to RigidBody as a child
4. Check CollisionShape2D disabled feature
5. Play

### PR DESCRIPTION
**Pull request by staddy**
Fixes #24183

### CODE REVIEW

File: servers/physics_2d/body_2d_sw.cpp [hunk 1]
```diff
@@ -61,17 +61,19 @@ void Body2DSW::update_inertias() {
 
 			for (int i = 0; i < get_shape_count(); i++) {
 
-				const Shape2DSW *shape = get_shape(i);
+				if (!get_shape_disabled(i)) {
```
Comment (akien-mga) [inline servers/physics_2d/body_2d_sw.cpp]: You could save one level of indentation with:
```
if (get_shape_disabled(i)) {
	continue;
}
```
and keep the rest unchanged.

**Comment by girng** [comment]
very nice

File: servers/physics_2d/collision_object_2d_sw.h [hunk 2]
```diff
@@ -115,6 +115,10 @@ class CollisionObject2DSW : public ShapeOwner2DSW {
 	void set_shape_metadata(int p_index, const Variant &p_metadata);
 
 	_FORCE_INLINE_ int get_shape_count() const { return shapes.size(); }
+	_FORCE_INLINE_ bool get_shape_disabled(int p_index) const {
```
Comment (akien-mga) [inline servers/physics_2d/collision_object_2d_sw.h]: I would name this `is_shape_disabled`.

**Comment by reduz** [comment]
Fix is good, can you apply the same code to the Godot 3D part of physics?

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**